### ISSUE TITLE
Crux-llvm counterexamples involving NaN

### ISSUE THREAD

**Issue by robdockins** [GaloisInc/crucible#188]
When constructing counterexamples involving `NaN`, `crux-llvm` outputs incorrect C syntax, (example below) which fails to compile.

```
float const crucible_values_float[] = { 3.746995e19,3.6894614e19,NaN,NaN,NaN,0.0,0.0,0.0,0.0,0.0,0.0 };
```

Some (but maybe not all?) versions of `math.h` include the `NAN` constant.  We should figure out what to do here.

**Comment by atomb** [comment]
I can think of two ways to potentially handle this. The simplest might be to include the following at the beginning of each counterexample file:

````
#include <math.h>
#define NaN NAN
````

A separate possibility could be to use `isNaN` in Haskell to test values before printing and, if a value is NaN, print out the actual bit representation. This seems more accurate if we can manage it, since it would allow us to distinguish different NaN values, but trickier to make work correctly. I ran a simple experiment where I constructed a Haskell `Float` value that passed `isNaN` in Haskell, used `unsafeCoerce` to cast it to a `Word32`, printed its hex value, pasted it  into a C file, and cast it to a `float`. The resulting value caused `isnan` in C to return false, so something in that pipeline must have changed its bit pattern.

**Comment by robdockins** [comment]
In C, casting from an integer type to a float will attempt to cast the numeric value, not the bitpattern.  AFAIK, directly casting the bitpattern requires either intrinsics of some sort, or using a `union` type.  I don't believe either can appear in an initializer list, unfortunately.

**Comment by atomb** [comment]
Given that, the first, simpler approach seems like the way to go. The worst thing that could happen is that it'd fail to compile on some strange platform that doesn't support "quiet" NaN values, and then we could figure out how to handle things on that platform.

In my Googling, I haven't come across a single mention of a specific platform that doesn't support them, so I don't know how rare that is. And [this page](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1011.htm) suggests making signaling NaNs optional, and quiet the default.

**Comment by atomb** [comment]
Going in to try and fix this, I noticed that it's already fixed, as of 4ec946c1927.

**Comment by robdockins** [comment]
Oh, oops, I really should have closed this at the time...
